protocol/synthetix/contracts/modules/core/IssueUSDModule.sol

```diff
@@ -96,8 +100,14 @@ contract IssueUSDModule is IIssueUSDModule {
         // Decrease the credit available in the vault
         pool.recalculateVaultCollateral(collateralType);
 
+        AssociatedSystem.Data storage usdToken = AssociatedSystem.load(_USD_TOKEN);
+
         // Mint stablecoins to the sender
-        AssociatedSystem.load(_USD_TOKEN).asToken().mint(msg.sender, amount);
+        usdToken.asToken().mint(address(this), amount);
+
+        account.collaterals[usdToken.getAddress()].increaseAvailableCollateral(
+            CollateralConfiguration.load(usdToken.getAddress()).convertTokenToSystemAmount(amount)
+        );
```

Comment:
Is this line here just to ensure we revert if we loose precision?

Reply:
yes but maybe Im too careful and we can save gas here